filetests/isa/x86/legalize-br-table.clif

```diff
@@ -16,8 +16,9 @@ ebb0(v0: i64):
 ; nextln:    brif uge $oob, ebb2
 ; nextln:    fallthrough $(inb=$EBB)
 ; check:   $inb:
+; nextln:    $(final_idx=$V) = uextend.i64 $idx
 ; nextln:    $(base=$V) = jump_table_base.i64 jt0
-; nextln:    $(rel_addr=$V) = jump_table_entry.i64 $idx, $base, 4, jt0
+; nextln:    $(rel_addr=$V) = jump_table_entry $final_idx, $base, 4, jt0
```

Comment:
Does this test still work with 64bit ints on i686?

Reply:
Line 3 in this file defines the target: target x86_64, so it should be fine, right?

Reply:
I mean if the target is set to `x86`, does it still work. Otherwise `cranelift_frontend::Switch` is broken for 64bit ints on 32bit archs.

Reply:
I think it *should* fail, since there's no way the compiler can decide what's the right thing to do, on a 32-bit platform, when seeing a 64-bits integer passed for a 32-bit operation. Now the question is how violentlym it fails... I think it will say there's no encoding for the instructions `br_table` is legalized to, and not panic at least. (There are likely many other failures on x86 32 bits, since it's untested at the moment.)

Reply:
Can you please change

https://github.com/CraneStation/cranelift/blob/05d22df6cbf98a072d7525cff4599d930af142a7/cranelift-frontend/src/switch.rs#L225

So that `cranelift_frontend::Switch` keeps working for 64bit integers on 32bit archs?

Reply:
I'd like to help, really, but I'm afraid I can't do this: there's no sensible reasonable default for what to do in this case. When an i64 is outside the uint32 range and is passed as the discriminant value of a br_table, we can either silently cut the high bits (above the uint32_max value), or we should signal an error because we're not doing what the developer might expect. It's up to the developer to do the right thing here.